core/trino-main/src/main/java/io/trino/connector/DynamicCatalogManagerModule.java

```diff
@@ -32,12 +36,17 @@ protected void setup(Binder binder)
         if (buildConfigObject(ServerConfig.class).isCoordinator()) {
             binder.bind(CoordinatorDynamicCatalogManager.class).in(Scopes.SINGLETON);
             CatalogStoreConfig config = buildConfigObject(CatalogStoreConfig.class);
-            switch (config.getCatalogStoreKind()) {
-                case MEMORY -> binder.bind(CatalogStore.class).to(InMemoryCatalogStore.class).in(Scopes.SINGLETON);
-                case FILE -> {
+            switch (config.getCatalogStoreKind().toLowerCase(Locale.ROOT)) {
+                case "memory" -> binder.bind(CatalogStore.class).to(InMemoryCatalogStore.class).in(Scopes.SINGLETON);
+                case "file" -> {
                     configBinder(binder).bindConfig(FileCatalogStoreConfig.class);
                     binder.bind(CatalogStore.class).to(FileCatalogStore.class).in(Scopes.SINGLETON);
                 }
+                default -> {
+                    binder.bind(CatalogStoreManager.class).in(Scopes.SINGLETON);
+                    newOptionalBinder(binder, CatalogStoreManager.class).setBinding().to(CatalogStoreManager.class).in(Scopes.SINGLETON);
```

Comment:
this binds `CatalogStoreManager` to itself - did you mean to introduce an interface for the `CatalogStoreManager`? If not this line can simply be removed since the actual binding is done just the line above.